tools/sdk-ai-bots/azure-sdk-qa-bot-evaluation/_evals_runner.py

```diff
@@ -0,0 +1,343 @@
+                            traceback.print_exc()
+                    else:
+                        outputFile.write(line)
+            outputFile.flush()
+            outputFile.close()
+
+    async def _call_bot_api(self, question: str, bot_endpoint: str, access_token: str, tenant_id: str = None, with_full_context: bool = True) -> Dict[str, Any]:
+        """Call the completion API endpoint."""
+        headers = {
+            "Content-Type": "application/json; charset=utf8",
+        }
+        if access_token: 
+            headers["Authorization"] = f"Bearer {access_token}"
+        
+        payload = {
+            "tenant_id": tenant_id if tenant_id is not None else "azure_sdk_qa_bot",
+            "message": {
+                "role": "user",
+                "content": question
+            },
+            "with_preprocess": True,
+            "with_full_context": with_full_context
+        }
+
+        async with aiohttp.ClientSession() as session:
+            async with session.post(bot_endpoint, json=payload, headers=headers) as resp:
+                if resp.status == 200:
+                    return await resp.json()
+                else:
+                    raise Exception(f"API request failed with status {resp.status}")
+
+    async def _prepare_dataset(self, testdata_dir: str, file_prefix: str = None, retrieve_response: bool = True):
+        """
+        Process markdown files in the data directory and generate Q&A pairs.
+        
+        Args:
+            prefix: Optional prefix to filter which markdown files to process.
+                        If provided, only files starting with this prefix will be processed.
+        """
+        logging.info("📁 Preparing dataset...")
+        data_dir = Path(testdata_dir)
+        logging.info(f"📂 Data directory: {data_dir.absolute()}")
+        script_directory = os.path.dirname(os.path.abspath(__file__))
+        logging.info(f"Script directory:{script_directory}")
+        output_dir = Path(os.path.join(script_directory, "output"))
+        output_dir.mkdir(exist_ok=True)
+        logging.info(f"📂 Output directory: {output_dir.absolute()}")
+        
+        for filename in os.listdir(output_dir.absolute()):
+            file_path = os.path.join(output_dir.absolute(), filename)
+            try:
+                if os.path.isfile(file_path) or os.path.islink(file_path):
+                    os.unlink(file_path)  # remove file or link
+                elif os.path.isdir(file_path):
+                    shutil.rmtree(file_path)  # remove directory
+            except Exception as e:
+                print(f"Failed to delete {file_path}. Reason: {e}")
+        print(f"Directory '{output_dir.absolute()}' cleared.")
+        
+        if not data_dir.exists():
+            logging.error(f"❌ Data directory {data_dir.absolute()} does not exist!")
+            return None
+        
+        current_date = datetime.now().strftime("%Y_%m_%d")
```

Comment:
Better to add README.md for this folder so that human could understand this project quickly. And it will be also useful if we want to leverage AI to add feature for this project in the future.